### ISSUE TITLE
No Permanent In-Game Identifications, Nor Bans For Serial Griefers From Low Pop Context

### ISSUE THREAD

**Issue by DougLefelhocz** [jasonrohrer/OneLife#529]
The griefer, [mentioned here](https://github.com/jasonrohrer/OneLifeData7/issues/619) and also in this [forum post](https://onehouronelife.com/forums/viewtopic.php?id=7059) (unless it's a frame job, which seems unlikely... will try to check later), started destroying things again and this time decided to make a set of connected pine floors and wood floors which spells out their discord name.  I talked to Jord about this individual, and I think Jord can reveal their discord name, and their in-game hash is traceable as the person was [Eve Leaf](http://lineage.onehouronelife.com/server.php?action=character_page&id=5872637) on Friday 02-Feb-2020, as confirmed by others.  This individual did more low pop griefing apparently on several servers in the past also, and in several towns where I played or saw in the past.

The griefer, so far as anyone knows, never has provided any in-game benefit whatsoever to any of us, as when a town gets stale for us, we can and have usually just started a new one, or as a sort of variation on our gaming experience like when three of us did a triplet Eve run this weekend, and then other people joined us and helped out.

City walls could block out that person from griefing a town.  But, there's a few problems which make such an idea unviable as a solution:

1. The server I play on tries to have an (unspoken) open sort of philosophy, at least according to my read, and I haven't heard anyone say anything in contrast to that.  I've logged into some other servers and gotten killed quickly as an Eve appearing in some people's fortress, or gotten cursed as a baby, but that's never happened on the server that I play on.  City walls _with locked doors_ aren't consistent with that philosophy.  

2. Towns have almost always, if not always, meant to eventually get connected to other towns via roads so that people who want to come back to them can do so fairly easy.  With a griefer on the loose trying to destroy things when one of us goes to sleep or do other things in life, such roads become questionable at best.  City walls don't work well with towns connected by roads.

3. The biggest problem lies in that, due to the griefer, it becomes wise to check every baby to see if they're a member of the existing discord.  This makes it unworkable to accept in new players and welcome them.  Or to accept people who just want to visit, as I've seen happen before.  Someone, who isn't known to anyone, doesn't have discord or doesn't have their password?  Sorry, they'll have to get excluded, because it ends up too probable that such a person will be the griefer.  The result of that in the long run doesn't end up good for veteran players, because they don't want to take the time to do that sort of thing unless they have to, and some new person ends up all too likely to just not get fed.  Or innocent players end up cursed, or the griefer gets cursed, and then soon enough becomes an Eve and can go back to destroying things.

Since servers 2-15 aren't likely at all to always have someone playing on them, it's likely that the only realistic solution would be to ban such a serial griefer from playing in such a context.  That such a person provides positive benefit to the bigserver context has not gotten proven to my knowledge, but that perhaps digresses from this issue.

**Comment by DougLefelhocz** [comment]
Update: The person tried to impersonate me yesterday, by using 'Eve Spoon' as his/her name and did appear on server12 again yesterday.  Just in case there's some confusion, my hash is 9a0589cb673446fd534812d43cde1913f30b256a while the griefer's hash is different.

**Comment by jasonrohrer** [comment]
So you're saying, essentially, that you have the tools with which to deal with this griefer (locked doors, for example), but that you don't want to use them?

In general, I'm looking for ways to let players deal with trouble in the game.  Having me serve as cop, judge, and jury is never going to work long-term.  Sifting through the logs for evidence.  He said she said.  It's an endless nightmare.

Obviously, the current curse system doesn't work at all in this context.


One thing to consider might be wandering far away from everything to start a town.  That way another lone Eve won't be able to find it, and with the Eve respawn mechanics, you'll be able to find it over and over.  That doesn't help with babies coming in, of course.  Nor does it help to let friend Eves play in that town when you're not there yourself.

Maybe you could Quad Eve, and then all walk in the same random direction, start a town together, and then have each of your Eve spawns set there in the future (so that you have four possible "seeds" when some of you want to play again there).

**Comment by DougLefelhocz** [comment]
> So you're saying, essentially, that you have the tools with which to deal with this griefer (locked doors, for example), but that you don't want to use them?

No, I don't have the tools to deal with this griefer.  I'd like to keep children in general, AND for everyone else to also keep their children in general also (or at least consider such if they are worried about their child growing up to do bad things)... those who want to stay of course.  But, with this serial griefer as potentially my child or someone else's it ends up too much of a risk to keep children who we don't know who they are.  It's a public server and people who want to join without talking to any of us in the past have been welcomed in the past and I'd like to welcome them in the future.  We had someone new start up last week, and got a player who said he was from China like two days ago (who knows if that person plays again on server12), and I've personally had the new player popup a few times for my children also (could have been the same person a few times, I didn't check logs for hashes).  Visitors from other servers have joined us before also.

Additionally, theoretically, the griefer could in principle do an apocalypse from some hard to find area in the vicinity of the tutorial structures.

> That doesn't help with babies coming in, of course. Nor does it help to let friend Eves play in that town when you're not there yourself.

Both of those are serious deficiencies.

The Quad Eve idea still leaves towns unconnected to the road network, doesn't resolve the issue of what to do about babies, and just suggests that players come basically by invite only.  But, it's a public server.  Again, at least myself if not most other people, try to be welcoming to visitors or people trying out server12.  Of course, someone otherwise unknown person could wander in and cause mayhem.  That's the risk of playing on a public server.  But, people deserve the presumption of innocence.  However, when it's a known serial griefer, and there don't exist tools that can deal with such a person while also being welcoming to players new to the server or visiting, it's not a happy situation.

**Comment by DougLefelhocz** [comment]
On top of this, with current spawning mechanics, it's possible that the griefer logs on when the server is empty, and then we have to wait however long it is until the griefer becomes infertile.

**Comment by jasonrohrer** [comment]
Curses don't work in low-pop.

Maybe if they did, that would help you to never have the griefer as your baby in the future.  However, there's also other risks in low-pop that you would be sent to d-town by a griefer.  Probably a bad idea.

Sounds like what you really want here is a private, invite-only server, but you're trying to play on a public server.

The public servers aren't designed with mechanics to keep certain people out, right?

In order for me to ban this guy, I have to believe your report (or spend time investigating).

But if I open that door, what is to stop this guy from telling stories to me that convince me to ban you?

Maybe in low-pop, personal curses should be soft, where the person can't go to d-town, but they are marked for you.  So you can flag people who are causing trouble and then avoid them in the future.

**Comment by jasonrohrer** [comment]
Apocalypse can't be done solo, due to the need for a bloody knife, right?

**Comment by DougLefelhocz** [comment]
> Sounds like what you really want here is a private, invite-only server, but you're trying to play on a public server.

No, that's not what I want.  I do like playing with people I haven't invited.  As I said before we had a new person just log in and start playing with us, no one that I know of had invited her, and she's fun to play with.  I just don't want a known serial griefer to destroy things.

> The public servers aren't designed with mechanics to keep certain people out, right?

No.  DonkeyTown exists, and the person ended up there after, from what I've heard, getting stabbed yesterday by someone else (the birth co-ordinates were (-21834,-20953)).

> In order for me to ban this guy, I have to believe your report (or spend time investigating).

> But if I open that door, what is to stop this guy from telling stories to me that convince me to ban you?

Permanent in-game identifications wouldn't have this issue.  Also, other people have stories about this person griefing them.  I don't think that other people would say that I've griefed their low pop towns before.

> Apocalypse can't be done solo, due to the need for a bloody knife, right?

All of the crafting for apocalypses can get done solo.  A single individual can wait near dark nosajs with a knife for when a baby gets born.  I would guess that when JCWilk had I think three, if not four endstones, and what I think was two completed tower stages in DonkeyTown, he did something like that.

**Comment by jasonrohrer** [comment]
Well, not out at the tutorial area, though, right?  Tutorial players are infertile.  Unless they build a landing strip out there and then come in there as a fertile Eve.

Identifying other players specifically is against the spirit of the game, of course.

The cursing mechanics allow you to mark a class of players, but not identify individuals.  However, this doesn't work on low-pop because of the danger of sending innocent people to donkeytown.

**Comment by jasonrohrer** [comment]
Hold the phone here...


I just tested it, and personal curses work fine in low-pop.

So just curse this griefer, and they will be marked for 30 days.

They will NEVER be your baby during this 30-day period, so you don't need to worry about that.

And they will never spawn as Eve near you, nor be born as a baby near you if you're playing with friends who don't have them cursed.  If you're near the only people currently playing, they will go to d-town.


This is the way for you to identify this person permanently, or at least for 30 days.


To avoid them spawning as Eve during off-hours and causing trouble, you and whoever trust each other must walk off into the distance and make a secret network of towns far away.

Now, if the griefer IS online currently, they will still be your mother.  So grow up and then kill them, right?  Twin together as the griefer's babies, and then form a posse at age 12 with bows to kill them.

**Comment by willowdean** [comment]
They tend to play when none of us are on or quickly SIDS before we can name and curse them. If they are our mother they will not raise us either. 
They are known to target people on our server and continually harass them, a few times leaving not the friendliness of notes in their town. In June/July they targeted the user betsst, who wrote the forum post, this month they targeted another user on our server, and now they have moved on to Doug. It’s not so much griefing to grief, it’s targeting users who they dislike/don’t agree with, at least on server12.

**Comment by DougLefelhocz** [comment]
> Identifying other players specifically is against the spirit of the game, of course.

This sounds rather bizarre for a few reasons:

1. You played on a multi-person Twitch stream with people before.  You specifically were able to identify other players during this time.

2. It's been a longstanding practice to ask children for their names.  The main issue lies in that it won't work reliably to identify someone like this griefer.  It does have benefits for the child sometimes also, since after someone has said their name, I've carried that baby to their town before, and also gotten carried to my town before.

3. The Boots lineage that had the longstanding overall lineage record and that still stands as the lineage record on server7, as I've heard, did identify people before letting them stay (there was a griefer trying to end their lineage prematurely).

**Comment by Dodg-e** [comment]
If it's on stream obviously you can identify someone...

**Comment by DougLefelhocz** [comment]
Yeah... I don't think streaming is against the spirit of the game.  At the very least, I only knew about this game because of Twitch streamers, and have logged in specifically to play with some of them before.  I would think of it as promotion.

**Comment by D3mon1cblack** [comment]
''Now, if the griefer IS online currently, they will still be your mother. So grow up and then kill them, right? Twin together as the griefer's babies, and then form a posse at age 12 with bows to kill them.'' so now we have to hope the griever is dumb enough to raise kids so we can kill the griever? in what world would a person that doesnt care about the people on the server care about raising kids?

**Comment by DougLefelhocz** [comment]
> If you're near the only people currently playing, they will go to d-town.

What happens if I curse the griefer, then come back after the end of that life, and they're the only person playing?  Will I end up in D-Town then and the griefer remains in the main area?

**Comment by jasonrohrer** [comment]
No, only if the griefer curses you will you end up in d-town.

Yes, obviously there are ways to route around the anonymity inherent in the premise of the game, through Discord, streams, playing with friends, etc.  But I'm not going to build "gamer tags" into the game itself.

If they SIDS before you name them, CURSE MY BABY will work for a short duration.  I'll fix it to work for longer.

**Comment by RosieRandom** [comment]
Today I was waiting until the server was empty (Art was on) so I could use my Eve spawn and not have to travel.  When I joined the server I was born to a newborn Eve who immediately cursed me.

Seems like I wasn't the only person monitoring the server population, except this person had a much worse reason to do so.  When Art logged in after me they were stabbed by the Eve.

I can't stop them from damaging any towns and anytime they are the only one on, I will get sent to donkey town.

This is my lineage link, my mother is the one that auto cursed and murdered
http://lineage.onehouronelife.com/server.php?action=character_page&id=5890181

**Comment by Artemissia** [comment]
I am the Art that Rosie is speaking of and I can verify that as soon as I was born I was stabbed.  I haven't tried to log back in with the griefer on since there is really nothing I can do at this point.  I bought the game about a month ago and have enjoyed playing on the low pop and big server but my favorite is the low pop because it is more of a community there.  Usually without the constant drama of the big server.  I ask is there no way that you can help with this issue?

**Comment by DougLefelhocz** [comment]
I logged in, as [Eve Going ](http://lineage.onehouronelife.com/server.php?action=character_page&id=5890344) after this person was no longer fertile and still on the server as an Eve.  I found dead sheep in four towns, including adobe rubble where a kiln had been and another kiln stood previously.  I found the grave of that person about 5 minutes after she had died.

**Comment by jasonrohrer** [comment]
Yeah, that is kindof a logjam, if the griefer jumps in and monopolizes Eve on a low-pop server for a while.

Of course, on the other side, if you have the griefer cursed, you WANT them sent to d-town, so they won't be born as your BB.


So I think the current way around this is to have all the "in" people run off and set their Eve spawns in the same location, far away in a secret place.

Then Grieve (griefer Eve) won't ever be able to spawn there.

First, if Grieve is your BB, you can curse them and then kill them, and then they will never be your BB again.

Second, if Grieve is cursed and born, they will be born far away in d-town.

Third, if Grieve spawns when no one else is on, then they will be at their own Eve spawn and won't be able to find your secret town.


There's still the problem of them blocking you while playing as Eve.  Either they are your mother (bad), or they have you cursed and you go to d-town.


But there really is no equitable way to deal with this, right?  You don't want to play with them, and they don't want to play with you.  You want them banned, but I'm guessing they would want you banned too...

The problem is that the server can't tell the difference between you and Grieve.  All it sees is that you both have each other cursed.


The only true solution here is a private server that you can control access to.  Then you can invite who you want and remove Grieve at will.

Actually, a server with a blacklist would be good enough, I suppose.  Open to public, but then people can be removed.

Maybe I will try to set up a "moderated" server as an experiment.  Who controls access, though?  The discord mods?

Seems like it would be a never-ending nightmare of he said she said, but if you really want that....

**Comment by DougLefelhocz** [comment]
"You want them banned, but I'm guessing they would want you banned too"

I don't think this at all likely.  The griefer put an abbreviation of their discord name (again, it's a person that Jord banned from the OHOL discord recently) in the town that I named with a sign that says 'Everton' which, I think, you could still go an observe.  Also, she put her name on bigserver2 according to an image that I saw, from someone in the low pop's discord, who hasn't commented here.  Thus, the griefer wants for us to know that she did such (I say 'she', since the OHOL character was female).  Also, I think it likely that the griefer wants to keep on destroying things others have built.  I would thus doubt that such a person would want us banned, since they wouldn't be able to destroy things we made then.

**Comment by Dodg-e** [comment]
Low pop could have saved spawn points?

Like /home would save current location for next spawn as Eve?

If you made a /home you spawn as Eve automatically

And to remove the saved location /removehome and making another /home would replace the other one?

Since low pop servers are made for a different type of game experience i dont see the issue with being able to spawn like this imo

Just needs to avoid being exploited by main servers when server restart for example and population is low (someone could get born to a specific point on main servers)

**Comment by Dodg-e** [comment]
But then griefer could spawn there too...

Maybe if someone is cursed and tries to make a /home near someone who cursed them they couldn't

**Comment by Tarr-OHOL** [comment]
Can't have a secret town Jason because logs exist of where you're born and die on a server. The only possible way to have an unreachable town is to do some very high level shenanigans such as flying to the tutorial then having a bunch of people spawn in the tutorial on said server to force the tutorial cells to shift too far out of walking distance. Not a very good solution at all in my opinion. 

I guess you could spread the Eve spawns on the server really far to discourage people from outside your play group from ever being near you but I think these guys like having (friendly) neighbors.


You can't lock your play area up due to things such as permanent blocking tiles existing because you might to to bed and find an ancient wall in its place (remember people gotta sleep and work, I'd say most people here are adults.) 

Even without the issue of permanent blocking structures you can't use base locks and keys because most people playing on these servers are going to have a search function on their client so they just have to ride/walk around until the big old flashing button says "Your key is right behind this tree!" 


These low pop players really do need something special to protect their way of play as when someone griefs BS2 it's annoying but when some asshole specifically goes to an off server to grief small group efforts it's a whole different level.

**Comment by Dodg-e** [comment]
1. Make a main town with plane and landing strip make secret town with landing strip
2. Get born to main town
3. Fly to secret town
4. Fly back to main town before you die
5. Hide plane

Seems pretty "easy"

**Comment by jasonrohrer** [comment]
Even if I were to ban this person, what would stop them from just buying a second account to subvert it?

I mean, if I did set up a moderated server, where the discord mods had power to blacklist accounts, the same thing would happen.

Notice that this is NOT an issue on BS2, which is the case the game is designed around.  This problem is solved on high-pop servers.

This game wasn't really designed around the low-pop case.

**Comment by jasonrohrer** [comment]
Tarr, isn't "their way of play" actually a private server where they get to pick who plays there?

They're asking to pick who plays with them on a public server that has no access controls as part of its design.

You could imagine all kinds of stuff like codes to play with friends in far-flung places on the server, and so on.  But in the end, all these things end up turning the experience into a private server experience.

Private servers are readily available for 5E per month:

https://zap-hosting.com/en/one-hour-one-life-server-hosting


Now, you might say you LIKE friendly random people who stumble in from time to time.  But the problem is that you don't know who's stumbling in, and if you allow randoms, you allow everyone, even the bad randoms.

**Comment by DougLefelhocz** [comment]
"I guess you could spread the Eve spawns on the server really far to discourage people from outside your play group from ever being near you but I think these guys like having (friendly) neighbors."

We do!

"Tarr, isn't "their way of play" actually a private server where they get to pick who plays there?"

As I've tried to say before, no.  With a private server someone has to get the name from some person who knows the name, and it's not public information.  So, people already know each other to some extent before they play with each other on a private server.  With your public servers Jason, someone just has to know the names of a particular server and how to log into one.  Two of the people above, in particular I've met in-game first, and then talked to them on the discord later.

"Now, you might say you LIKE friendly random people who stumble in from time to time. But the problem is that you don't know who's stumbling in, and if you allow randoms, you allow everyone, even the bad randoms."

Jason, you're talking about interacting with people initially.  Sure, there's a risk of dealing with a bad random initially, and I think that risk, in particular, worth it.  You met a stranger in the real world, you never know how they will be after all.  And in my opinion, rebuilding one time because of a griefer isn't very bad.  

However, it's different when someone is known to grief over and over and over again ad nauseum. And such a person has a history of doing such from a while ago.  And as predicted before by someone other me, apparently has intention to target many other, if not every town.

And the vast majority of random people are fine.  Really, out of something probably exceeding 3 dozen people where I've played with people I haven't known before in a low pop situation, there's just this one person I've known who has been a serial griefer like this.

Also, people have budgets in the real world.

"Notice that this is NOT an issue on BS2, which is the case the game is designed around."

People on the discord talk about, from what I've heard, a longtime serial playerkiller known as Ziv.  Also, I saw a lot of boilers in Twisted's stream, and as I said before the name of the player I referred to initially put an abbreviation of their in-game name according to a screenshot I saw.  The person who posted the screenshot has played on server12 for like 2 weeks, and has no known previous experience with that individual to my knowledge.  I seriously doubt that serial griefing is fixed on bs2.

**Comment by Kinrany** [comment]
Soo would the problem be solved if it was possible to curse the griefer as a baby?

**Comment by jasonrohrer** [comment]
That's already possible.  Curse my baby.  Even after they /DIE

The problem is that they will have you cursed too, which means that if they are currently Eve, they are blocking things up, because they are sending you to d-town, or killing you as a bb if you are born to them.

Finally, it sounds like it's very easy for them to find your town and destroy it (due to server logs that are published on a 24-hour delay).


Doug, what you are requesting is a public server where you are king and you can decide to banish people.  Or you're asking me to be king and believe you and banish this person on your behalf.

I hope you can understand that I don't have time to "investigate" such cases.  I would have to count on you to be trustworthy, and just go with what you're telling me.

But what's to stop the griefer from telling me similar tales about you?  And then I trust them and banish you?

How do I know that you're not the griefer all along, trying to frame your victim?

And even if you are truthful, how do we know that some other griefer hasn't framed the person whom your asking to be banned?


Ziv may have killed a lot of players, but now you (all of you) can decide not to play near Ziv for 30 days simply by cursing him one time.

These issues are possible to solve on BS2 because there are enough active players for a quorum.  If all of them curse Ziv, then ziv goes to d-town, while Ziv doesn't have enough power alone to send anyone to d-town (b/c there are loads of places for them to get born, and Ziv is only in one of those places).

On server12, what we're dealing with is essentially a 1-v-1 situation.  Griefer is trying to get you, and you are trying to "get" the griefer by banning them.  There is no quorum.  Just two people who disagree.

There may be more than two people involved long-term, but short-term, the Griefer is currently Eve, and you want to be Eve.

If you are born as a baby, the Griefer will kill you.

But what happens when the Griefer is born as YOUR baby?  You kill them too, right?

You block them from playing, just as they block you from playing.  Who's server is it anyway?  You want it to be yours.

**Comment by DougLefelhocz** [comment]
> I hope you can understand that I don't have time to "investigate" such cases. I would have to count on you to be trustworthy, and just go with what you're telling me.

> But what's to stop the griefer from telling me similar tales about you? And then I trust them and banish you?

> How do I know that you're not the griefer all along, trying to frame your victim?

In this case, there are other people who can verify that such a person has been a serial griefer.  On the main OHOL discord, and have commented above.

I said this at 3:05 P. M. in the discord: "Jason apparently thinks that serial griefing is fixed on bs2... I have a hard time believing that"  A few veterans responded and didn't think serial griefing fixed also.

> On server12, what we're dealing with is essentially a 1-v-1 situation. Griefer is trying to get you, and you are trying to "get" the griefer by banning them. There is no quorum. Just two people who disagree.

Willowdean, Artemis, and Rosie have commented above.  There exist others in the discord who've seen things.  No, this isn't just two people who disagree.

> There may be more than two people involved long-term, but short-term, the Griefer is currently Eve, and you want to be Eve.

That wasn't the case when I logged in this morning and had Rosie as my child.

> But what happens when the Griefer is born as YOUR baby? You kill them too, right?

If I can detect that such a person is the griefer very soon, I don't feed the baby and let such a baby die from starvation, not murder.

> You want it to be yours.

No, that's not what I want.  I also wouldn't consider someone popping into one of 'my' towns and developing it as griefing also.  I also put the word 'my' in quotes as it only refers to me or someone else having the primary Eve spawn in that location, or that I've been the main person to develop such a place, or was the person who started it.  'My' doesn't refer to ownership.  For example, I once started up a new place.  A day or two later another Eve came in, and developed it.  Another person, who made a forum post looking for low pop people also joined, and the town got a joint name.

**Comment by Kinrany** [comment]
@DougLefelhocz Please use `>` for quoting, this isn't a FluxBB forum 😝

**Comment by DougLefelhocz** [comment]
I checked what happened yesterday using the [name log](http://publicdata.onehouronelife.com/publicLifeLogData/lifeLog_server12.onehouronelife.com/2020_02February_05_Wednesday_names.txt) and [the text log](http://publicdata.onehouronelife.com/publicLifeLogData/lifeLog_server12.onehouronelife.com/2020_02February_05_Wednesday.txt).  I can reveal the griefer's hash here if desired, but have been reluctant to do so due to a previous incident.  The player with that hash had two children, both of whom died at 0.64, the second child dying due to a killer, matching Roise and Art's description above and what happened in this [lineage](http://lineage.onehouronelife.com/server.php?action=character_page&id=5890181).  Additionally, the two children of Eve Going match the children of that Eve in the same order, as reported above.

Anyone else could confirm these details using the logs.

I think the above thus indicates another way that there exists serious evidence that I'm not the griefer here.

**Comment by jasonrohrer** [comment]
Doug, I said I don't have time to investigate such matters.  I can't hear testimony from witnesses or check logs.  It might work this time, to get rid of this one guy, but what about next time, and the time after that?  I've been down this road before, and it never ends.

What it boils down to is you and the current griefer disagreeing about how Server 12 should be used.  You think your way is the right way, and the griefer's way is the wrong way.  You want them banished, but they probably also want YOU banished (for bringing the matter to my attention, at least, or just to get your goat).

This is what I mean about you wanting it to be "your" server.  You want to have a final say over who is forbidden from playing there (with support of your comrades, of course).

But, you know, it's a public server.  It's a bathroom wall, and you're trying to draw a big square on the wall and say, "NO ONE ELSE CAN WRITE IN THIS SQUARE OR CROSS OUT WHAT I WRITE HERE."

Then, when someone else writes in your square, and you don't like what they write, you appeal to the owner of the restaurant to investigate using handwriting recognition, security cameras, witnesses, etc, and eventually banish the guilty party from the bathroom.


The reality is that the low-pop servers are a vestigial edge case.  They are there with hopes and dreams of this game ever getting popular enough to fill them all (3000 simultaneous players, or whatever it would take).  They are not there to serve as a low-pop experience.  They should probably be removed to save money (and will eventually, long term), but for now, they are cheap enough to keep around.  Also, because the dream of 3000 players isn't dead just yet.

But if Pew-Die-Pie ever played OHOL, Server 12 would be overrun by 80+ players just like all the other servers.


For the time being, I can't think of a way to "solve" this problem beyond me investigating and manually banning someone from this server.  I'm not going down that road.


Regarding BS2, while nothing is "solved" as in QED, I think the problem is solvable without manual investigation and banning, and it's pretty close to solved currently.  A 30-day personal area ban on an individual really does prevent them from bothering you in the future.

Groups can also maintain property fences and leadership chains long-term to keep track of who they trust, and to protect their infrastructure.  The healing mechanics help to insulate groups against a lone murderer, and the posse mechanics help a group track down and kill a lone trouble-maker.

But these things ONLY function in a persistent multiplayer context where we have a lot of people and a "quorum" working against whoever the quorum considers to be a griefer.


They don't, obviously, work 1-v-1, where there's no one to follow you, no one to inherit your property gate, no one to heal you, and no one to join your posse.  It's just your word against their word.  The "good guy" is not detectable, from the server's perspective.


I'm open to suggestions about how to solve this problem automatically (without manual investigation), but the logician in me suspects that no solution exists, given the 1-v-1 nature of the problem, and inherent moral relativism.


Github Issues is not a place to petition for banning.

You have your permanent (30-day) in-game identification through the curse system.  You and your comrades should curse this person.  I will make the "curse my baby" window a bit longer (currently 5 seconds, I think).

**Comment by jasonrohrer** [comment]
Here's an example (partial) solution:

If there's only one mother around, and you have her curse-blocked OR she has you curse-blocked, you don't get born to her, and instead spawn as a new Eve.

This will at least prevent the griefer from blocking you from playing.

But it also will prevent YOU from blocking the griefer from playing.  They will spawn as Eve nearby instead of in d-town.


UNLESS you have more than one mother active on the server, and you all have the griefer curse-blocked---then the griefer will go to d-town.

But beware that the griefer could do the same thing to you if they get a comrade.

**Comment by DougLefelhocz** [comment]
> What it boils down to is you and the current griefer disagreeing about how Server 12 should be used. 

Jason, you advertise OHOL as a multiplayer game of parenting and civilization building.  Isn't that what you want players how to think of the game also, with some freedom for interpretation as to what exactly that means?  At least sometimes?  Above the griefer has murdered a child instantly, after starving the previous child, with no knowledge of who either person was.  Do you really think that making bunch of boilers, doors without any walls to put them on, bow drill bows, rabbit snares never moved near rabbit holes, chopping a bunch of trees which structured pens, burning all tanks of kerosene in a town on a car or plane sitting there, killing all animals in pens, and basically the equivalent of graffiti of their name, and then also going to other towns and killing all of their sheep, qualifies as civilization building?  Also, Twisted saw on stream the name 'Fish' which was the graffitied name in my town, and a bunch of boilers in that bigserver2 town also.

Do you really want your servers used so that those who build up towns have everything wrecked by someone who just wants to destroy things, and there's not a realistic way to stop them?  If so, how can the person wrecking things be said to play within the spirit of the game?

> The reality is that the low-pop servers are a vestigial edge case. 

A bunch of boilers on bigserver2 and the same name there as in Everton.  Also, that's a rather strange thing to say as the reality every week is that there exists a window where bigserver2 is low pop in a few respects, and additionally, server1 has some similarities.  Were the player population to drop below 15, and iirc Kinrany has worried about such long-term, and doesn't dropsy no longer exist on bigserver2?  Tool slots didn't exist on bigserver2 when I was Eve Volek, and I saw on Eve Bones's stream (which got removed from his channel, I believe due to a technical problem) that his character didn't have tool slots either.  Power failures can and have occurred in the real world due to causes that people didn't understand at the time or natural forces.  On top of that, you have a "today's deep roots".  There's been days where that would have been empty if currently low pop servers didn't exist.  Also, I, and others have had the new player popup in the low pop context.  I'm sure that some players have preferred playing some of their early lives in that context than thrown in with the bigserver2 context.

Do you really believe that no substantial similarities ever exist between the low pop context and bigserver2, and that bigserver2 will never have any similarities to low pop servers, especially after the recent problems from steam during updates?

> Regarding BS2, while nothing is "solved" as in QED, I think the problem is solvable without manual investigation and banning, and it's pretty close to solved currently. A 30-day personal area ban on an individual really does prevent them from bothering you in the future.

This all assumes that the griefer is caught by people who know OHOL's mechanics curse them in time.  One griefer posted images on the discord of chopping down what looked like all of the rubber trees in a jungle (who would catch such a person if they tried to do so again?).  [This video](https://www.youtube.com/watch?v=tBf4UM6QQTo) appeared last night (that individual isn't often thought of as a serial griefer).  This [video](https://www.youtube.com/watch?v=tybcV60UcuM) yesterday.  These aren't people standing around talking or casually roleplaying (which I don't want to classify as griefing).  At least during those lives, they're trying to work against people trying to build civilization and/or parent and/or maintain civilization, aren't they?  Someone will probably think up more ways to do destructive things in the future.

> I'm open to suggestions about how to solve this problem automatically (without manual investigation), but the logician in me suspects that no solution exists, given the 1-v-1 nature of the problem, and inherent moral relativism.

So whether someone plays in the spirit of a game intended to be about civilization building and/or parenting with other people is just a morally relativistic matter where anything goes?  I couldn't disagree more.  Sometimes things are just plain wrong.

But, hey it's your game.  I can only try to make you aware of how I think things work, even when that involves concepts that one would think you would want your game to fulfill.

**Comment by jasonrohrer** [comment]
The point, Doug, is that I'm not going to manually investigate anything that goes on inside the game, ever.

I've gone down that road before with other games, and I'm not going to go down that road here.  It's a never-ending nightmare of he-said/she-said, false accusations, feigned innocence, angry emails from banned people, and HOURS on my part spent investigating each case.

Look at this thread as an example of hours wasted.


So, the policy is:  whatever happens inside the game stays in the game.  There is no "higher justice" than what's possible inside the game.

Just like, you know, real life.

You can't ban someone from real life.  You can kill them or build a fence to keep them out.

At least in this game, you can go beyond real life and curse them, but that's only there because the game has a non-real-life feature of reincarnation.  Dead criminals stay dead in real life, but not in this game.

The game is advertised as "a multiplayer survival game of parenting and civilization building".

Part of "civilization building" is figuring out how to deal with destructive actors.  Not "me" figuring that out by deciding who to ban, but "you" figuring that out using the tools that I've given you.  Tools which only work in a multiplayer context, as advertised.

The game is not advertised as an oasis of solo-play, semi-private servers where I swoop in to solve your 1-v-1 disputes for you as policeman, judge, and jury.

Criminality is defined as what the group finds objectionable.  But in a solo play context, there is no group.  If two players are on an island together, and they disagree, which one is the criminal?


I gave you the tools to build a jury yourself, inside the game.  But a "jury of one" is just a dictatorship.  I didn't promise to give you the tools to do that.


Regarding "the spirit of the game," the only spirit present is what is embodied by the mechanics themselves.  Playing "against the spirit" of Chess or Go is impossible, despite what some people have said in the past when people employed "perverse" strategies.

**Comment by jasonrohrer** [comment]
And like I said, if you have automatable solutions beyond "investigate this person and ban them," I'm an eager audience.

**Comment by jasonrohrer** [comment]
Please open a new issue if you have an automatable solution, beyond the two fixes that I implemented this week.

**Comment by DougLefelhocz** [comment]
> So, the policy is: whatever happens inside the game stays in the game. There is no "higher justice" than what's possible inside the game.  Just like, you know, real life.

No, I don't know that about real life.  Some people believe in a higher power which decides such matters in real life.  Some others believe in karma.  And even people who regard those as superstitions may well expect that people in the future might look back at past matters and have some relevant moral judgment about people who have died.

> The game is not advertised as an oasis of solo-play, semi-private servers where I swoop in to solve your 1-v-1 disputes for you as policeman, judge, and jury.

Low pop servers are not semi-private servers, and server12 hasn't been semi-private for much of the day this week.  There's 5 people on it right now as I write this.

How did three people comment above, if this is a 1v1 dispute?  I probably also should have said so earlier, but I sent you an invite to the relevant discord which we use (a discord server I do NOT have ownership over, nor seek to have ownership over).

> Criminality is defined as what the group finds objectionable. But in a solo play context, there is no group. 

Again, this is not a solo play context here Jason.

> I gave you the tools to build a jury yourself, inside the game.

No, you did not.  Players start as babies on the server as long as they remain low pop.  Cursing isn't even relevant until the player can curse the player's name and potentially name that player.  Players can run far out into the woods on bigserver2.  Additionally, another similarity between the low pop context and what happens when players shift from bigserver2 to server1 during the update period, where there exist a small handful of players left on bigserver2 sometimes for what seems like 10 minutes.

> Regarding "the spirit of the game," the only spirit present is what is embodied by the mechanics themselves.

Then the game can't ever have any exploits, though I seem to remember you using that term before.  Players wouldn't ever be able to play incorrectly, no matter how much you might dislike what they are doing, since they always play in line with what is embodied by the mechanics themselves.  If you consistently believed that, you would have to always be perfectly content with everything that players have done.

> And like I said, if you have automatable solutions beyond "investigate this person and ban them," I'm an eager audience.

Serial griefers can always be more inventive than solutions which can get automated.  Good and intelligent people are necessary to deal with them.  But since everything in the game is inline with what gets embodied by the mechanics themselves, there's apparently not even been griefing in the game, with griefing just being whatever is against the spirit of the game.  Just an empty term that people use, but nothing that people can prove as definitively wrong to a hyptheoretical objective observer.  I suspect that intelligent people would just walk away from calling anything griefing in the game in some sort of provable sense with the spirit of the game just being embodied by the mechanics themselves.

**Comment by DougLefelhocz** [comment]
> The reality is that the low-pop servers are a vestigial edge case. They are there with hopes and dreams of this game ever getting popular enough to fill them all (3000 simultaneous players, or whatever it would take). They are not there to serve as a low-pop experience. They should probably be removed to save money (and will eventually, long term), but for now, they are cheap enough to keep around. Also, because the dream of 3000 players isn't dead just yet.

The oldest OHOL video that I know of on Twitch is [this video](https://www.twitch.tv/videos/425012426)  You act like it doesn't mean much, but if so, why did LowImpact take steps to preserve that video?  Why haven't any of the bigserver2 or main server players apparently taken such steps?

**Comment by kuwas** [comment]
@jasonrohrer What seems clear to me, is that there are a lot of players who aren't satisfied with the default play style, as prescribed by you on the official servers... this seems to be a recurring theme.

So instead of trying to address or not address specific requests, why not just add a custom server discovery service, like in Counter-Strike, Rust, Dont Starve Together, etc. Players already have the ability to run their own servers, so having a discovery service will allow other players to choose the server with the style they want. Of course this service should have a big disclaimer that you are not responsible for what happens on non-official servers.

**Comment by Kinrany** [comment]
> So instead of trying to address or not address specific requests, why not just add a custom server discovery service, like in Counter-Strike, Rust, Dont Starve Together, etc.

That would be Jason shooting himself in the foot. This is totally something the community should do.

That, or access to the server browser could also be a paid service bundled with the game...

**Comment by kuwas** [comment]
@Kinrany You do know that custom servers can use the same ticket servers for authentication right?

**Comment by Kinrany** [comment]
Yes, but they won't. And I don't think it's possible to only allow the servers that do.

**Comment by DougLefelhocz** [comment]
> Regarding "the spirit of the game," the only spirit present is what is embodied by the mechanics themselves. Playing "against the spirit" of Chess or Go is impossible, despite what some people have said in the past when people employed "perverse" strategies.

Were what you claim the case for Chess, then not trying to play in such a way to draw or checkmate your opponent, would be consistent with the spirit of chess, since chess mechanics allow it.  A person just moving pawns and other pieces legally with full knowledge of the end conditions of a chess game, but no attempt to draw or win, would just as much play in the spirit of chess as someone trying to checkmate an opponent.  All the chess leagues in the world wouldn't measure the quality of chess players, and so many artificial intelligence projects in chess wouldn't have ever meant a thing.  Similar reasoning applies to Go.

And really, worse.  Someone just looking to annoy or pester someone else playing Chess or Go, and who could do so, would still be playing in the spirit of Chess or Go.

**Comment by kuwas** [comment]
> Yes, but they won't. And I don't think it's possible to only allow the servers that do.

Of course its possible, just do a two way authentication with a trusted third party, the ticket server.

1. Client authenticates to the Ticket server
  1a. Client sends their license and a challenge
  1b. If the Client sent an invalid license, Ticket server sends back an error response
  1c. If the Client sent a valid license, Ticket server sends back an encrypted token, with an auth token and the challenge inside
2. Client sends the encrypted token to the Game server
3. Game server send the encrypted token to the Ticket server, for decryption
  3a. If the Game server sent an encrypted token with an invalid auth token, Ticket server sends back an error response
  3b. If the Game server sent an encrypted token with a valid auth token, Ticket server sends back the client challenge
4. Game server sends the client challenge to the Client
  4a. If the challenge is incorrect, reject the server
  4b. If the challenge is correct, accept the server

**Comment by Kinrany** [comment]
You can't trust the third-party game server...

**Comment by kuwas** [comment]
> You can't trust the third-party game server...

I don't understand what you're talking about, both client and server are communicating with the official ticket server, the goal here is to check license. In terms of privacy, the player's email and license are never sent to the game server, only an auth token and generated challenge.

In terms of users trusting the game server, they already do this at their own risk today, with custom servers. Users also do this in many other games out there, at their own discretion, thats why I said there needs to be a disclaimer.

**Comment by DougLefelhocz** [comment]
Fish identified herself in-game yesterday as [Eve Fish](http://publicdata.onehouronelife.com/publicLifeLogData/lifeLog_server12.onehouronelife.com/2020_02February_06_Thursday_names.txt).  Fish's hash is 08378da6608ea3eea3ca42c921547a45d77d82e6.  A plaster wall building with doors on it had the doors replaced with stone walls, which have now turned ancient.

There's more from what I'm hearing, but I'll reserve details in case what I've read ends up someone else.

**Comment by Kinrany** [comment]
Why would the game server communicate with the auth server at all?

**Comment by kuwas** [comment]
> Why would the game server communicate with the auth server at all?

🤦‍♂

**Comment by DougLefelhocz** [comment]
I've sent another discord invite.

**Comment by unoriginalartist** [comment]
Have you guys forgotten you can say curse my baby?
